odoo/sql_db.py

```diff
@@ -545,7 +606,7 @@ def autocommit(self, on):
     def commit(self):
         """ Perform an SQL `COMMIT` """
         self.flush()
-        self._cursor.execute('SAVEPOINT "%s"' % self._savepoint)
+        self._savepoint = self._cursor.savepoint(flush=False)
```

Comment:
You should close the former savepoint object (without rollback), so that it is released.

Reply:
Doesn't work as explained in the commit message, there's a test which doesn't work at all if I do that, and I ended up giving up after spending a few hours trying to understand why.

Reply:
Can you please leave a `TODO` comment, then?
I like the idea of "lazy" savepoints for test cursors, by the way.
We should dig that idea in some future task.